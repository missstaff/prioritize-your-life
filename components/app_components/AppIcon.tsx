import React from "react";
import { View } from "react-native";
import { ScaledSheet, s } from "react-native-size-matters";
import Ionicons from "@expo/vector-icons/Ionicons";
import { type IconProps } from "@expo/vector-icons/build/createIconSet";
import { type ComponentProps } from "react";

/**
 * Renders an app icon component.
 *
 * @param style - The style object for the component.
 * @param rest - Additional props for the component.
 * @returns The rendered app icon component.
 */
export function AppIcon({
  style,
  ...rest
}: IconProps<ComponentProps<typeof Ionicons>["name"] | any>): JSX.Element {
  return (
    <View style={[styles.container, style]}>
      <Ionicons size={s(28)} {...rest} />
    </View>
  );
}

const styles = ScaledSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: "transparent",
    borderRadius: 100,
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    padding: s(2.5),
  },
});
