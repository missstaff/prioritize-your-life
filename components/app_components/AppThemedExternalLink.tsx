import { Href, Link } from "expo-router";
import { Platform } from "react-native";
import { openBrowserAsync } from "expo-web-browser";
import { type ComponentProps } from "react";
import { useThemeColor } from "@/hooks/useThemeColor";
import { COLORTHEME } from "@/constants/Colors";

type Props = Omit<ComponentProps<typeof Link>, "href"> & {
  href: Href<string | object>;
};

const AppThemedExternalLink = ({ href, style, ...rest }: Props) => {
  const color = useThemeColor(
    { light: COLORTHEME.light.background, dark: COLORTHEME.dark.background },
    "background"
  );
  return (
    <Link
      target="_blank"
      {...rest}
      href={href}
      onPress={async (event) => {
        if (Platform.OS !== "web") {
          // Prevent the default behavior of linking to the default browser on native.
          event.preventDefault();
          // Open the link in an in-app browser.
          await openBrowserAsync(href as string);
        }
      }}
      style={[{ color }, style]}
    />
  );
};

export default AppThemedExternalLink;
