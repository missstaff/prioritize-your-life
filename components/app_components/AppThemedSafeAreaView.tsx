import { COLORTHEME } from "@/constants/Colors";
import { useThemeColor } from "@/hooks/useThemeColor";
import { SafeAreaView } from "react-native";


const AppThemedSafeAreaView  = ({ children }: { children: React.ReactNode }) => {
    const backgroundColor = useThemeColor(
        { light: COLORTHEME.light.background, dark: COLORTHEME.dark.background },
        "background"
      );
    return (
        <SafeAreaView style={{flex: 1, backgroundColor: backgroundColor}}>
            {children}
        </SafeAreaView>
    );
};
export default AppThemedSafeAreaView;