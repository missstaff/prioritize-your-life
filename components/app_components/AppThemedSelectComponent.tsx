
const AppThemedSelectComponent = () => {};

export default AppThemedSelectComponent;