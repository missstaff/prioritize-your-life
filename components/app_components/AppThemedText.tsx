import { Text, type TextProps } from "react-native";
import { ScaledSheet, s } from "react-native-size-matters";
import { useThemeColor } from "@/hooks/useThemeColor";
import { COLORS, COLORTHEME } from "@/constants/Colors";

export type ThemedTextProps = TextProps & {
  type?: "default" | "title" | "defaultSemiBold" | "subtitle" | "link";
};

const AppThemedText = ({
  style,
  type = "default",
  ...rest
}: ThemedTextProps) => {
  const color = useThemeColor(
    { light: COLORTHEME.light.text, dark: COLORTHEME.dark.text },
    "text"
  );

  return (
    <Text
      style={[
        { color },
        type === "default" ? styles.default : undefined,
        type === "title" ? styles.title : undefined,
        type === "defaultSemiBold" ? styles.defaultSemiBold : undefined,
        type === "subtitle" ? styles.subtitle : undefined,
        type === "link" ? styles.link : undefined,
        style,
      ]}
      {...rest}
    />
  );
};

const styles = ScaledSheet.create({
  default: {
    fontSize: s(16),
    lineHeight: s(24),
  },
  defaultSemiBold: {
    fontSize: s(16),
    fontWeight: "600",
    lineHeight: s(24),
  },
  title: {
    fontSize: s(32),
    fontWeight: "bold",
    lineHeight: s(32),
  },
  subtitle: {
    fontSize: s(20),
    fontWeight: "bold",
  },
  link: {
    color: COLORS.primary,
    fontSize: s(16),
    lineHeight: s(30),
  },
});

export default AppThemedText;
