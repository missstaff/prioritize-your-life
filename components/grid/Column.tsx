import { ScaledSheet } from "react-native-size-matters";
import AppThemedView from "../app_components/AppThemedView";

interface ColumnProps {
  children: React.ReactNode;
  inlineStyles?: object;
}

const Column = ({ children, inlineStyles, ...props }: ColumnProps) => {
  return (
    <AppThemedView style={[styles.column, inlineStyles]} {...props}>
      {children}
    </AppThemedView>
  );
};

const styles = ScaledSheet.create({
  column: {
    alignItems: "flex-start",
    display: "flex",
    flexDirection: "column",
    flexGrow: 1,
    justifyContent: "center",
    width: "33%",
  },
});

export default Column;
