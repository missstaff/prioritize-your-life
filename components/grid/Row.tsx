import { ScaledSheet, s, vs } from "react-native-size-matters";
import AppThemedView from "../app_components/AppThemedView";

interface RowProps {
  children: React.ReactNode;
}

const Row = ({ children, ...props }: RowProps) => {
  return (
    <AppThemedView style={styles.row} {...props}>
      {children}
    </AppThemedView>
  );
};

const styles = ScaledSheet.create({
  row: {
    alignItems: "center",
    display: "flex",
    flexDirection: "row",
    justifyContent: "flex-start",
    paddingHorizontal: s(5),
    paddingVertical: vs(5),
    width: "100%",
  },
});

export default Row;
