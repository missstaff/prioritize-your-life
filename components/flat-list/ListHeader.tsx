import AppThemedText from "../app_components/AppThemedText";
import Column from "../grid/Column";
import Row from "../grid/Row";
import { TextStyle, ViewStyle } from "react-native";

export interface ListHeaderProps {
  headings: string[];
  colStyles?: ViewStyle;
  rowStyles?: ViewStyle;
  textStyles?: TextStyle;
}

const ListHeader = ({ headings, colStyles, textStyles, rowStyles  }: ListHeaderProps) => {
  return (
    <Row style={rowStyles}>
      {headings.map((heading, index) => (
        <Column colStyles={colStyles} key={index}>
          <AppThemedText style={textStyles} type="defaultSemiBold">
            {heading}
          </AppThemedText>
        </Column>
      ))}
    </Row>
  );
};


export default ListHeader;
