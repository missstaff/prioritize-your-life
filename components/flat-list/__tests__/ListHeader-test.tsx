import React from "react";
import { render } from "@testing-library/react-native";
import ListHeader, { ListHeaderProps } from "../ListHeader";
import { AppThemedText } from "../../app_components/AppThemedText";
import Column from "../../grid/Column";
import Row from "../../grid/Row";

// Mocking dependencies
jest.mock("../../grid/Row", () => jest.fn(({ children }) => <div>{children}</div>));
jest.mock("../../grid/Column", () => jest.fn(({ children }) => <div>{children}</div>));
jest.mock("../../app_components/AppThemedText", () => ({
  AppThemedText: jest.fn(({ children }) => <div>{children}</div>),
}));

describe("ListHeader Component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  const mockStyles = {
    fontSize: 16,
    color: "black",
  };

  const headings = ["Heading 1", "Heading 2", "Heading 3"];

  it("renders correctly with headings", () => {
    const props: ListHeaderProps = {
      styles: mockStyles,
      headings,
    };

    const { toJSON } = render(<ListHeader {...props} />);

    expect(toJSON()).toMatchSnapshot();
  });

  // #TODO fix this test
//   it("renders the correct number of headings", () => {
//     const props: ListHeaderProps = {
//       styles: mockStyles,
//       headings,
//     };

//     const { getByText } = render(<ListHeader {...props} />);

//     // Check if all headings are rendered
//     headings.forEach((heading) => {
//       expect(getByText(heading)).toBeTruthy();
//     });

//     // Check that Column was called for each heading
//     expect(Column).toHaveBeenCalledTimes(headings.length);

//     // Check that AppThemedText was called for each heading
//     expect(AppThemedText).toHaveBeenCalledTimes(headings.length);
//   });

  it("passes the correct styles to AppThemedText", () => {
    const props: ListHeaderProps = {
      styles: mockStyles,
      headings,
    };

    render(<ListHeader {...props} />);

    expect(AppThemedText).toHaveBeenCalledWith(
      expect.objectContaining({
        style: mockStyles,
      }),
      {}
    );
  });
});
