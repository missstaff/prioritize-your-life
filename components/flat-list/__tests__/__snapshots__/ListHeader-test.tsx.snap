// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeader Component renders correctly with headings 1`] = `
<div>
  <div>
    <div>
      Heading 1
    </div>
  </div>
  <div>
    <div>
      Heading 2
    </div>
  </div>
  <div>
    <div>
      Heading 3
    </div>
  </div>
</div>
`;
